src/plugins/commander.go

```diff
@@ -71,10 +74,57 @@ func (c *Commander) Process(msg *core.Message) {
 			c.sendCommand(c.ctx, cmd)
 		case core.AgentConnected:
 			c.agentRegistered(cmd)
+		case core.AgentConfig:
+			c.agentBackoff(cmd)
 		}
 	}
 }
 
+func (c *Commander) agentBackoff(cmd *proto.Command) {
+	log.Infof("agentBackoff in commander.go with command %v ", cmd)
+
+	if cmd.GetAgentConfig() == nil {
+		log.Warnf("update commander client with default backoff settings as agent config nil, for a pause command %+v", cmd)
+		c.cmdr.WithBackoffSettings(client.DefaultBackoffSettings)
+		return
+	}
+	if cmd.GetAgentConfig().GetDetails() == nil {
+		log.Warnf("update commander client with default backoff settings as agent details nil, for a pause command %+v", cmd)
+		c.cmdr.WithBackoffSettings(client.DefaultBackoffSettings)
+		return
+	}
+	if cmd.GetAgentConfig().GetDetails().GetServer() == nil {
+		log.Warnf("update commander client with default backoff settings as server nil, for a pause command %+v", cmd)
+		c.cmdr.WithBackoffSettings(client.DefaultBackoffSettings)
+		return
+	}
+	backoffSetting := cmd.GetAgentConfig().GetDetails().GetServer().Backoff
+	if backoffSetting == nil {
+		log.Warnf("update commander client with default backoff settings as backoff settings nil, for a pause command %+v", cmd)
```

Comment:
Is this missing actually setting the backoff settings back to default?
```suggestion
		log.Warnf("update commander client with default backoff settings as backoff settings nil, for a pause command %+v", cmd)
		c.cmdr.WithBackoffSettings(client.DefaultBackoffSettings)

```